src/controllers/controller.bar.js

```diff
@@ -284,7 +285,14 @@ module.exports = function(Chart) {
 		 * @private
 		 */
 		getIndexScale: function() {
-			return this.getScaleForId(this.getIndexScaleId());
+			var scale = this.getScaleForId(this.getIndexScaleId());
+			if (scale.options.categoryPercentage === undefined) {
+				scale.options.categoryPercentage = defaultBar.scales.xAxes[0].categoryPercentage;
```

Comment:
[nit] I would put `defaultBar.scales.xAxes[0]` into a local variable to save a few bytes